### DISCUSSION TITLE
PMD not building PMD-CLI and PMD-Distribution packages

### DISCUSSION THREAD

**Discussion by NathanMacDiarmid** [pmd/pmd#4735] [Q&A] [answered]
I have cloned the PMD source code to my local machine and tried building it using both this terminal command ./mvnw clean verify and from the Maven side panel on VSCode (running clean and verify separately. Both of them build successfully, however, it does not build either the PMD-CLI package or the PMD-Distribution package, resulting in not outputting a .zip file of PMD under pmd/pmd-dist/target. Any suggestions on how we can fix this?
![Capture](https://github.com/pmd/pmd/assets/62114750/b9ade4e5-c3bd-45e6-b5ec-086ae5207b8a)

**Comment by juanitarodeloo** [comment]
I am having the same problem, we are both working on a project that is using PMD.
![PMDNotBuilding](https://github.com/pmd/pmd/assets/122914651/f164b6ad-54d2-44fc-b972-52bb13df5e63)

**Comment by adangel** [comment, marked as answer]
This is currently a side effect of the cyclic dependency between PMD and PMD-Designer, see #4446 for details.

The two modules pmd-cli and pmd-dist have been extracted into a separate maven profile "cli-dist", which should have been activeByDefault:
https://github.com/pmd/pmd/blob/c5503c938fec727b8d2a108ccd70797b6a0c6c08/pom.xml#L1178-L1187

But it is obviously not always activeByDefault... Do you have any profiles activated in your settings.xml? That would explain it (sadly, a maven profile that is activeByDefault is only active until any other profile is activated explicitly).

In order to build pmd-cli and pmd-dist, you need activate the profile explicitly:

```
./mvnw clean verify -Pcli-dist
```

I hope to fix this issue before the final 7.0.0 release.

**Reply by juanitarodeloo** [reply]
That must be it, that worked for us. Thank you for getting back to us so quickly!
